### PR TITLE
dns: bypass EDNS query

### PR DESCRIPTION
**Pull request by botengyao**
Signed-off-by: Boteng Yao <boteng@google.com>

DNS filter will ignore the additional RRs as EDNS could have them, the filter will fall back to the traditional behavior.
Issue #22676.

Commit Message:
Additional Description:
Risk Level: Low
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: source/extensions/filters/udp/dns_filter/dns_parser.cc [hunk 1]
```diff
@@ -231,6 +229,13 @@ bool DnsMessageParser::parseDnsObject(DnsQueryContextPtr& context,
     }
     context->queries_.push_back(std::move(rec));
   }
+
+  // We could encounter additional RRs in an EDNS query, and Envoy will ignore them.
+  if (context->header_.additional_rrs) {
+    ENVOY_LOG(debug, "Ignoring additional RRs in a query because Envoy does not support. "
```
Comment (yanjunxiang-google) [inline source/extensions/filters/udp/dns_filter/dns_parser.cc:235]: Should we add a counter here, then in the test check the counter is increased if such query is received?

Comment (botengyao) [reply source/extensions/filters/udp/dns_filter/dns_parser.cc:236]: Done! Thank you!

**Comment by yanjunxiang-google** [comment]
LGTM

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**